### PR TITLE
Fix CI workflows.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
1. use `macos-12` instead of `macos-latest`
2. prepare to switch from `ubuntu-latest` to `ubunty-$ver`
3. rename few CIs for better understading their statuses
4. disable CD workflows on forks

### CODE REVIEW

**Review by jonathanl-bq — APPROVED** [review]

File: .github/workflows/csharp.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ concurrency:
     cancel-in-progress: true
 
 jobs:
-    run-tests:
+    build-and-test-c#-client:
```
Comment (acarbonetto) [inline .github/workflows/csharp.yml:34]: please split out the rename calls.  that way if AMZ wants to revert the -latest changes, they can revert the whole PR

File: .github/workflows/csharp.yml [hunk 2]
```diff
@@ -62,7 +62,7 @@ jobs:
             - name: Install shared software dependencies
               uses: ./.github/workflows/install-shared-dependencies
               with:
-                  os: ${{ matrix.os }}
+                  os: ${{ contains(matrix.os, 'macos') && 'macos' || 'ubuntu' }}
```
Comment (acarbonetto) [inline .github/workflows/csharp.yml:65]:
```suggestion
                  os: ${{ contains(matrix.os, 'macos') && ('macos' || 'ubuntu') }}
```

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -53,7 +53,7 @@ jobs:
             - name: Install shared software dependencies
               uses: ./.github/workflows/install-shared-dependencies
               with:
-                  os: ${{ matrix.os }}
+                  os: ${{ contains(matrix.os, 'macos') && 'macos' || 'ubuntu' }}
```
Comment (acarbonetto) [inline .github/workflows/go.yml:56]:
```suggestion
                  os: ${{ contains(matrix.os, 'macos') && ('macos' || 'ubuntu') }}
```

File: .github/workflows/java.yml [hunk 4]
```diff
@@ -60,7 +60,7 @@ jobs:
             - name: Install shared software dependencies
               uses: ./.github/workflows/install-shared-dependencies
               with:
-                  os: ${{ matrix.os }}
+                  os: ${{ contains(matrix.os, 'macos') && 'macos' || 'ubuntu' }}
```
Comment (acarbonetto) [inline .github/workflows/java.yml:63]:
```suggestion
                  os: ${{ contains(matrix.os, 'macos') && ('macos' || 'ubuntu') }}
```

File: .github/workflows/npm-cd.yml [hunk 5]
```diff
@@ -18,17 +18,18 @@ concurrency:
...
-          uses: actions/checkout@v4
-        - name: Start self hosted EC2 runner
-          uses: ./.github/workflows/start-self-hosted-runner
-          with:
-              aws-access-key-id: ${{ secrets.AWS_EC2_ACCESS_KEY_ID }}
-              aws-secret-access-key: ${{ secrets.AWS_EC2_SECRET_ACCESS_KEY }}
-              aws-region: ${{ secrets.AWS_REGION }}
-              ec2-instance-id: ${{ secrets.AWS_EC2_INSTANCE_ID }}
+        if: github.repository_owner == 'aws'
+        runs-on: ubuntu-latest
```
Comment (acarbonetto) [inline .github/workflows/npm-cd.yml:22]:
```suggestion
        runs-on: ubuntu
```

**Review by acarbonetto — COMMENTED** [review]
a couple of minor suggestions

File: .github/workflows/csharp.yml (hunk 2 above)
Comment (Yury-Fridlyand) [reply .github/workflows/csharp.yml:65]: nonono, it is actually
```
os: ${{ (contains(matrix.os, 'macos') && 'macos') || 'ubuntu' }}
```
do you want brackets?

File: .github/workflows/npm-cd.yml (hunk 5 above)
Comment (Yury-Fridlyand) [reply .github/workflows/npm-cd.yml:22]: no such runner

File: .github/workflows/csharp.yml (hunk 2 above)
Comment (acarbonetto) [reply .github/workflows/csharp.yml:65]: oh yes!  please!!!!

File: .github/workflows/npm-cd.yml (hunk 5 above)
Comment (acarbonetto) [reply .github/workflows/npm-cd.yml:22]: so the runner will probably have the same problem

Comment (Yury-Fridlyand) [reply .github/workflows/npm-cd.yml:22]: sure, will do another PR to fix bubuntu later

**Pull request merged**